### PR TITLE
remove deprecated JSON-RPC server

### PR DESCRIPTION
**Pull request by tersec**
https://github.com/status-im/nimbus-eth2/releases/tag/v22.3.0
> The [JSON-RPC](https://nimbus.guide/api.html) service (`--rpc` flag) option is now deprecated. It's scheduled for removal in version `v22.6` (i.e. June of this year). If you are currently relying on the JSON-RPC API, please consider switching to the official [REST API](https://nimbus.guide/rest-api.html).

### CODE REVIEW

File: beacon_chain/conf.nim [hunk 1]
```diff
@@ -337,23 +337,6 @@ type
         defaultValueDesc: ""
         name: "status-bar-contents" }: string
 
-      rpcEnabled* {.
```
Comment (arnetheduck) [inline beacon_chain/conf.nim:340]: probably better to keep the options (hidden) and log a warning - ie I've seen these options in copy-paste guides where it's doubtful the service is actually used, no point breaking those user setups just to make a point.

Comment (tersec) [reply beacon_chain/conf.nim:340]: https://github.com/status-im/nimbus-eth2/pull/3656/commits/25013b1ff7b56b8b54172059da09002af3e952b1

Comment (arnetheduck) [reply beacon_chain/conf.nim:340]: very nice, much better UX

**Review by arnetheduck — APPROVED** [review]

**Comment by arnetheduck** [comment]
The client in https://github.com/status-im/nimbus-eth2/tree/mitem/beacon_chain/spec/eth2_apis can also be removed - likewise validator exit support etc - this would also need corresponding updates to the manual.

**Comment by tersec** [comment]
https://github.com/status-im/nimbus-eth2/pull/3676

**Pull request merged**